src/AppBundle/Entity/User.php

```diff
@@ -170,9 +172,13 @@ public function getId()
     /**
      * @return Department
      */
-    public function getDepartment(): Department
+    public function getDepartment()
     {
-        return $this->getFieldOfStudy()->getDepartment();
+        $fos = $this->getFieldOfStudy();
+        if ($fos !== null) {
```

Comment:
Er det ikke slik at alle brukere har en FieldOfStudy?

Reply:
Det stemmer, men akkurat i det man oppretter nye brukere så opprettes det `User`-entities som ikke har `FieldOfStudy`. 

Både når man søker
![image](https://user-images.githubusercontent.com/5937246/43849720-61a7cd5e-9b36-11e8-9b2c-005969b3d95b.png)

og når man oppretter nye brukere i kontrollpanelet
![image](https://user-images.githubusercontent.com/5937246/43849775-8210b09c-9b36-11e8-887b-01b467b82194.png)

Reply:
Vil problemes løses hvis det valideres at det må være satt en department/FoS?

Reply:
Du har helt rett. Vi validerer allerede at FoS er satt, men problemet lå i at emailvalidatoren prøvde å kjøre `$user->getDepartment()` for å finne ut om brukeren hadde søkt før, og dette skjedde før den fikk validert FoS.

Men det var jo helt feil sted å sjekke dette i `Entity\User`, når jeg kunne sjekket det direkte i `Service\ApplicationAdmission` slik jeg gjør det i nyeste commit. :smile: